samples/subsys/mgmt/mcumgr/smp_svr/README.rst

```diff
@@ -1,117 +1,271 @@
+   Intel Hex (``.hex``) image format. This is written as ``(bin|hex)`` in this
+   document.
 
-Using MCUboot's `imgtool.py` script, convert the `zephyr.bin` file from step 3
-into an image file.  In the below example, the MCUboot repo is located at
-`~/repos/mcuboot`.
+Using MCUboot's :file:`imgtool.py` script, sign the :file:`zephyr.(bin|hex)`
+file you built in Step 3. In the below example, the MCUboot repo is located at
+:file:`~/src/mcuboot`.
 
 .. code-block:: console
 
-    ~/repos/mcuboot/scripts/imgtool.py sign \
+   ~/src/mcuboot/scripts/imgtool.py sign \
+        --key ~/src/mcuboot/root-rsa-2048.pem \
         --header-size 0x200 \
         --align 8 \
         --version 1.0 \
         --included-header \
-        --key ~/repos/mcuboot/root-rsa-2048.pem \
-        <path-to-smp_svr-zephyr.bin> signed.img
+        <path-to-zephyr.(bin|hex)> signed.(bin|hex)
 
-The above command creates an image file called `signed.img` in the current
-directory.
+The above command creates an image file called :file:`signed.(bin|hex)` in the
+current directory.
 
-Step 5: Upload the smp_svr image
-================================
+Step 5: Flash the smp_svr image
+===============================
 
-Upload the `signed.img` file from step 4 to image slot 0 of your board.  The
-location of image slot 0 varies by BSP.  For the nRF52dk, slot 0 is located at
-address 0xc000.
+Upload the :file:`signed.(bin|hex)` file from Step 4 to image slot-0 of your
+board.  The location of image slot-0 varies by board, as described in
+:ref:`mcuboot_partitions`.  For the nRF52 DK, slot-0 is located at address
+``0xc000``.
+
+Using :file:`nrfjprog` you don't need to specify the slot-0 starting address,
+since :file:`.hex` files already contain that information:
+
+.. code-block:: console
+
+    nrfjprog --program <path-to-signed.hex>
+
+Using GDB:
 
-The following gdb command uploads the image to 0xc000:
 .. code-block:: console
 
-    restore <path-to-signed.img> binary 0xc000
+    restore <path-to-signed.bin> binary 0xc000
 
-Step 7: Run it!
+Step 6: Run it!
 ===============
 
-The `smp_svr` app is ready to run.  Just reset your board and test the app with
-the mcumgr CLI tool:
+.. note::
+   If you haven't installed :file:`mcumgr` yet, then do so by following the
+   instructions in the :ref:`mcumgr_cli` section of the Management subsytem
+   documentation.
+
+.. note::
+   The :file:`mcumgr` command-line tool requires a connection string in order
+   to identify the remote target device. In this sample we use a BLE-based
+   connection string, and you might need to modify it depending on the
+   BLE controller you are using.
+
+
+The ``smp_svr`` app is ready to run.  Just reset your board and test the app
+with the :file:`mcumgr` command-line tool's ``echo`` functionality, which will
+send a string to the remote target device and have it echo it back:
+
+.. code-block:: console
+
+   $ sudo mcumgr --conntype ble --connstring ctlr_name=hci0,peer_name='Zephyr' echo hello
```

Comment:
Please remove the leading `$`.
My understanding is that leading `$`s in copy-and-pasteable commands are being phased out, as mentioned in https://github.com/zephyrproject-rtos/zephyr/pull/6181#discussion_r167965594

Reply:
agreed, fixed